### ISSUE TITLE
Dashboard starts submodules in wrong directory

### ISSUE THREAD

**Issue by jimnz111** [microsoft/vscode-spring-boot-dashboard#71]
I have a webapplication in a submodule that uses XSL files from WEB-INF/lib. This uses the Spring XsltResolver class, and it is never able to find the files. This works flawlessley in Eclipse. Having done quite a lot of debugging of the issue, I have now found out the root cause. 

I've got an app with the following directory structure
```bash
c:\Users\eXXXXXXX\source\ccss-webapp
           - submodule1
           - submodule2
           - ccss_webapp
                    - src/main/webapp
                               WEB-INF/lib
          - pom.xml
```
The webapplication itself is in folder ccss_webapp, which is defined as a module in the root pom.xml (this is the parent pom)

The spring dashboard correctly shows me all the submodules, and I start the ccss_webapp just fine. The problem lies in the setup of the context root, which, if I debug carefully I find that it points to DocumentRoot.getValidDirectory(), which is a springframework file. This method will call getCommonDirectories(), and it iterates through 3 names to try and see if that folder is in the current working directory. Heres the files and code it uses to attempt to locate the running application directory:

```java
private static final String[] COMMON_DOC_ROOTS = { "src/main/webapp", "public",
			"static" };
...
	private File getCommonDocumentRoot() {
		for (String commonDocRoot : COMMON_DOC_ROOTS) {
			File root = new File(commonDocRoot);
			if (root.exists() && root.isDirectory()) {
				return root.getAbsoluteFile();
			}
		}
		return null;
	}
```
It attempts to see if "src/main/webapp" exists. Under eclipse it does, but under VSCode it doesn't. 

A quick debug tells me whats going on. If I sysout the same directory (src/main/webapp) to find out where I am in the directory tree this is what I get :
```java
System.out.println(new java.io.File("src/main/webapp").getAbsoluteFile())
```
```bash
(void)
c:\Users\eXXXXXXX\source\ccss-webapp\src\main\webapp
```
The directory is not correct because the application working directory is not within the module directory. It should be saying c:\Users\e1069023\source\ccss-webapp\ccss-webapp\

If I prepend the submodule name on then everything works just fine. Here in the same debug point I do just that.

```java
System.out.println(new java.io.File("ccss_webapp/src/main/webapp/").exists())
```
```bash
(void)
true
```

Its now correct and the application XSL rendering now works flawlessley. (I suspect a number of JSP issues I've seen raised on forums may be caused from the same). 

Taking the same approach as above, within a linux container mounted with the same source directory from Windows, if I start a spring-boot:run in source\ccss-webapp the XSL template resolution fails, but if I do the same from source\ccss-webapp\ccss_webapp\ directory it works perfectly

I read up on the documentation for launch,json and was able to resolve the issue quite simply by adding this to the configuration :

```bash
"cwd": "${cwd}/ccss_webapp",
```

And everything works perfectly now! Having said that I had to spend some time finding out why its not working, and I doubt a lot of developers would be able to work out exactly what is going on, and end up not using spring-boot-dashboard because of it, which isn't what we want because VSCode is so much better than Eclipse.

I aren't familiar with typescript and the startup modules in vscode-spring-boot-dashboard to suggest a fix to this. I assume its something in Controller.ts, but don't know where to start!
Can someone with more knowledge of this project suggest a fix for this?

**Comment by Eskibear** [comment]
So you mean, when you right click on ccss_webapp, then click run/debug, the app fails because of the wrong cwd. Exactly we didn't set a specific cwd, leaving the default value to the workspace root folder. 
I imagine that, if we set value of `cwd` to the folder of the project/submodule you click, the problem would get resolved?

Refs:
[launch.json](https://github.com/microsoft/vscode-spring-boot-dashboard/blob/cc9f52b618396eba4ebce27b7676597aeef2e6bf/src/Controller.ts#L69-L75)
vscode API definition [here](https://github.com/microsoft/vscode/blob/52f0c468ae2beaebe67222cc9df4191e36fb7e18/src/vs/vscode.d.ts#L9038-L9050)

**Comment by jimnz111** [comment]
Yes if it were set to default to project/submodule clicked this would resolve it. 

In hindsight now that I know how it works, its really trivial to fix it by updating launch.json, but it really may just hold some people back when something doesn't work ootb.

**Comment by Eskibear** [comment]
OK, I'll submit a fix changing it like below, and it should work ootb then.
```ts
        await vscode.debug.startDebugging(
            vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(app.path)),
            Object.assign({}, targetConfig, {
                noDebug: !debug,
                cwd: app.path,  <<<---add this line
                vmArgs
            })
        );
```

**Comment by Eskibear** [comment]
@jimnz111 Have a try on this private build [extension-vsix.zip](https://github.com/microsoft/vscode-spring-boot-dashboard/files/3449477/extension-vsix.zip), see if it works for your case.

**Comment by jimnz111** [comment]
This is fantastic :) I removed launch.json and selected submodule, and it correctly launched the spring boot app. 

Looking in launch.json it now has this in there for cwd.
```
            "cwd": "${workspaceFolder}",
```

Thanks for getting a fix together so quickly.

**Comment by Eskibear** [comment]
Thank you for verification. It will soon be shipped in v0.1.6.

**Comment by Eskibear** [comment]
>   "cwd": "${workspaceFolder}"

Actually `workspaceFolder` is the default cwd, and the value is `xxxxx\source\ccss-webapp` in your case. When you click on the submodule to start, the value of cwd is overrode to the folder containing its pom.xml, i.e. `xxxxx\source\ccss-webapp\ccss_webapp`.
If you launch the debug session directly in debug view, or <F5>, it still uses the original one in launch.json. (so in your case, the app would fail to run correctly)

It's a little bit tricky, just to be clear about the mechanism.
